Neuro/Minigames/Completion/Solvers/ReplaceWaterJugSolver.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Neuro.Cursor;
+using System.Collections;
+using UnityEngine;
+
+namespace Neuro.Minigames.Completion.Solvers;
+
+[MinigameSolver(typeof(WaterStage))]
+public sealed class ReplaceWaterJugSolver : MinigameSolver<WaterStage>
+{
+    public override IEnumerator CompleteMinigame(WaterStage minigame, NormalPlayerTask task)
+    {
+        yield return InGameCursor.Instance.CoMoveTo(minigame.waterButton.transform.position);
+        minigame.Refuel();
+        while (!minigame.complete)
+            yield return new WaitForFixedUpdate();
+        minigame.Close();
```

Comment:
Why do you call minigame.Close from here? Doesn't it get called automatically from the minigame itself?

Reply:
In my testing the minigame gets stuck trying to fill the tank even when it is full, presumably a side effect of calling Refuel() directly even though the Refuel function itself checks if the minigame is complete. Therefore a manual call to Close is required to close the window.

Reply:
A small delay would look a bit better

```suggestion
        yield return Sleep(0.2f);
        minigame.Close();
```

```suggestion
        yield return minigame.CoStartClose(0.2f);
```